controllers/integration/integration_adapter.go

```diff
@@ -137,6 +137,136 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
+		if applicationSnapshotEnvironmentBinding != nil {
+			applicationSnapshotEnvironmentBinding.Spec.Snapshot = applicationSnapshot.Spec.DisplayName
+			applicationSnapshotComponents := a.getApplicationSnapshotComponents(*components)
+			applicationSnapshotEnvironmentBinding.Spec.Components = *applicationSnapshotComponents
+		} else {
+			applicationSnapshotEnvironmentBinding, err := a.CreateApplicationSnapshotEnvironmentBinding(
+				bindingName, a.application.Namespace, a.application.Name,
+				availableEnvironment.Spec.DisplayName,
+				applicationSnapshot, *components)
+
+			if err != nil {
+				a.logger.Error(err, "Failed to create ApplicationSnapshotEnvironmentBinding",
+					"ApplicationSnapshotEnvironmentBindingName.Application", applicationSnapshotEnvironmentBinding.Spec.Application,
+					"ApplicationSnapshotEnvironmentBindingName.Environment", applicationSnapshotEnvironmentBinding.Spec.Environment,
+					"ApplicationSnapshotEnvironmentBindingName.Snapshot", applicationSnapshotEnvironmentBinding.Spec.Snapshot)
+				return results.RequeueOnErrorOrStop(a.updateStatus())
+			}
+
+			a.logger.Info("Created new ApplicationSnapshotEnvironmentBinding",
+				"ApplicationSnapshotEnvironmentBindingName.Application", applicationSnapshotEnvironmentBinding.Spec.Application,
+				"ApplicationSnapshotEnvironmentBindingName.Environment", applicationSnapshotEnvironmentBinding.Spec.Environment,
+				"ApplicationSnapshotEnvironmentBindingName.Snapshot", applicationSnapshotEnvironmentBinding.Spec.Snapshot)
+		}
+	}
+	return results.ContinueProcessing()
+}
+
+func (a *Adapter) getApplicationSnapshotComponents(components []hasv1alpha1.Component) *[]appstudioshared.BindingComponent {
+	bindingComponents := []appstudioshared.BindingComponent{}
+	for _, component := range components {
+		bindingComponents = append(bindingComponents, appstudioshared.BindingComponent{
+			Name: component.Spec.ComponentName,
+			Configuration: appstudioshared.BindingComponentConfiguration{
+				Env: []appstudioshared.EnvVarPair{
+					{Name: component.Spec.Env[0].Name,
+						Value: component.Spec.Env[0].Value,
+					},
+				},
+				Replicas: component.Spec.Replicas,
+			},
+		})
+	}
+	return &bindingComponents
+}
+
+func (a *Adapter) CreateApplicationSnapshotEnvironmentBinding(bindingName string, namespace string, applicationName string, environmentName string, appSnapshot *appstudioshared.ApplicationSnapshot, components []hasv1alpha1.Component) (*appstudioshared.ApplicationSnapshotEnvironmentBinding, error) {
+	bindingComponents := a.getApplicationSnapshotComponents(components)
+
+	applicationSnapshotEnvironmentBinding := &appstudioshared.ApplicationSnapshotEnvironmentBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bindingName,
+			Namespace: namespace,
+		},
+		Spec: appstudioshared.ApplicationSnapshotEnvironmentBindingSpec{
+			Application: applicationName,
+			Environment: environmentName,
+			Snapshot:    appSnapshot.Spec.DisplayName,
```

Comment:
We need to set the snapshot to point to the resource name, so it should be `applicationSnapshot.Name` instead of `applicationSnapshot.Spec.DisplayName`.

Reply:
Updated